.github/workflows/typos.yaml

```diff
@@ -1,29 +0,0 @@
-# Check pull requests for typos.
```

Comment:
Why are we removing this workflow?

Reply:
Isn't it used in the PR pipelines? Or is there another spell check that's configured there?

Reply:
If it's used, I can't find where. Maybe we should have something like that, though

Reply:
it's triggered by Github for all PRs:
<img width="696" alt="image" src="https://github.com/microsoft/chat-copilot/assets/52973358/d321e78f-28f8-4982-832c-9fb9cad1c5cb">

Reply:
@dehoward Thanks! I didn't think of PR's for some reason... Reverting this deletion.